### DISCUSSION TITLE
Error on production server

### DISCUSSION THREAD

**Discussion by wolverine4277** [spatie/pdf-to-image#234] [General]
Hi everybody!
In the development server this simple script to convert from PDF to image works fine:

```
<?php
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);

require __DIR__ . "/vendor/autoload.php";
putenv( 'PATH=' . getenv('PATH') . 'C:\Program Files\gs\gs10.03.1\bin');

use Spatie\PdfToImage\Exceptions\InvalidFormat;
use Spatie\PdfToImage\Exceptions\PageDoesNotExist;
use Spatie\PdfToImage\Exceptions\PdfDoesNotExist;
use Spatie\PdfToImage\Pdf;

$pathToPdf = __DIR__.'/files/qr.pdf';
$pathToWhereImageShouldBeStored = __DIR__.'/files/qr.jpg';
$pdf = new Spatie\PdfToImage\Pdf($pathToPdf);
$pdf->save($pathToWhereImageShouldBeStored);
?>
```
But when i try on production server the script fails, error 500, no errors on PHP log file. In IIS log the error is:

2024-06-20 13:49:48 XXX.XXX.XXX.XXX GET /spatie/test.php - 80 - XXX.XX.XXX.XXX Mozilla/5.0+(Windows+NT+10.0;+Win64;+x64)+AppleWebKit/537.36+(KHTML,+like+Gecko)+Chrome/126.0.0.0+Safari/537.36 - 500 0 258 300082

In both servers:
-Download ImageMagick PHP extension (8.3.3) having in mind the PHP version, the thread safety (NTS), and the architecture (x64). In my case https://downloads.php.net/~windows/pecl/releases/imagick/3.7.0/php_imagick-3.7.0-8.3-nts-vs16-x64.zip
-Extract from php_imagick-….zip the php_imagick.dll file, and save it to the ext directory of your PHP installation
-Extract from php_imagick-….zip the other DLL files (they may start with CORE_RL, FILTER, IM_MOD_RL, or ImageMagickObject depending on the version), and save them to the PHP root directory (where you have php.exe), or to a directory in your PATH variable
-Add this line to your php.ini file:
extension=php_imagick.dll
-Download and install Ghostcript (and rename the executable to gs.exe)

After run the script is that In the temp directory there are two temp files, the original PDF without extension and another of 1 byte size.
The problem is with the line 188 (pdf-to-image/src/Pdf.php):

`$imageData = $this->getImageData($path, $page->number);`

If a run this [script](https://stackoverflow.com/questions/70768399/test-imagick-image) all works fine.
I'm searching info in the web but i don't find a solution to this problem.
If anyone can help me with this problem, I would be very grateful.
Regards.

Sebastián

**Comment by patinthehat** [comment]
@wolverine4277 what version of the package are you using?

**Comment by wolverine4277** [comment]
Hi @patinthehat, is  the last one, I download the package yesterday.
I try before with other versión (2.5 I think) and same problem but in constructor with line that use readImage.

**Comment by patinthehat** [comment]
@wolverine4277 Take a look at [this comment](https://github.com/spatie/pdf-to-image/issues/217#issuecomment-2169122731) and let us know if it resolves your issue :+1:

**Comment by wolverine4277** [comment]
Hi @patinthehat , I already do that because the error named in the comment. Now I don't get any errores bit the script fails returning 500 error.

**Comment by patinthehat** [comment]
@wolverine4277 I believe your issue may be due to low resource limits specified in your `policy.xml` file for Imagemagick. Try increasing the memory, file, etc. limits there and then test again.

Imagemagick seems to default to segfaulting when it runs out of memory, which would be the reason there are no error messages/logs.

**Reply by wolverine4277** [reply]
Hi @patinthehat, thanks for the tip. I had no idea about that file, I'm going to check it tomorrow.
Regards.
